cms/models.py

```diff
@@ -934,9 +969,20 @@ def get_context(self, request, *args, **kwargs):
         # Hits a circular import at the top of the module
         from courses.models import CourseRunEnrollment
 
+        self.course = (
```

Comment:
Could we do this as a `cached_property` instead?

Reply:
self.course is basically the related course that was fetched using the model one to one relation field. Had to set that as we needed to do select related to reduce queries. Otherwise, it just fetches the course and makes extra queries for related objects. I have created a property with different name in https://github.com/mitodl/mitxpro/pull/2502/commits/b6bbe613b5434ce87d46185b73c2b90c8cb5adbe. Let me know if that looks good or if the old implementation seems better.